purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -268,6 +249,113 @@ class Purchases @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE) intern
         )
     }
 
+    private fun syncPurchaseWithBackend(
+        purchaseToken: String,
+        storeUserID: String?,
+        appUserID: String,
+        productInfo: ReceiptInfo,
+        onSuccess: () -> Unit,
+        onError: (PurchasesError) -> Unit,
+    ) {
+        val unsyncedSubscriberAttributesByKey =
+            subscriberAttributesManager.getUnsyncedSubscriberAttributes(appUserID)
+        backend.postReceiptData(
+            purchaseToken = purchaseToken,
+            appUserID = appUserID,
+            isRestore = this.allowSharingPlayStoreAccount,
+            observerMode = !this.finishTransactions,
+            subscriberAttributes = unsyncedSubscriberAttributesByKey.toBackendMap(),
+            receiptInfo = productInfo,
+            storeAppUserID = storeUserID,
+            onSuccess = { info, body ->
+                subscriberAttributesManager.markAsSynced(
+                    appUserID,
+                    unsyncedSubscriberAttributesByKey,
+                    body.getAttributeErrors()
+                )
+                deviceCache.addSuccessfullyPostedToken(purchaseToken)
+                cacheCustomerInfo(info)
+                sendUpdatedCustomerInfoToDelegateIfChanged(info)
+                onSuccess()
+            },
+            onError = { error, errorIsFinishable, body ->
+                if (errorIsFinishable) {
+                    subscriberAttributesManager.markAsSynced(
+                        appUserID,
+                        unsyncedSubscriberAttributesByKey,
+                        body.getAttributeErrors()
+                    )
+                    deviceCache.addSuccessfullyPostedToken(purchaseToken)
+                }
+                onError(error)
+            }
+        )
+    }
+
+    /**
+     * This method will send a purchase to the RevenueCat backend. This function should only be called if you are
+     * in Amazon observer mode or performing a client side migration of your current users to RevenueCat.
+     *
+     * The receipt IDs are cached if successfully posted so they are not posted more than once.
+     *
+     * @param [productID] Product ID associated to the purchase.
+     * @param [receiptId] ReceiptId that represents the Amazon purchase.
+     * @param [amazonUserID] Amazon's userID. This parameter will be ignored when syncing a Google purchase.
+     * @param [isoCurrencyCode] Product's currency code in ISO 4217 format.
+     * @param [price] Product's price.
+     */
+    fun syncObserverModeAmazonPurchase(
+        productID: String,
```

Comment:
can we normalize on ID being fully uppercase?

Reply:
oh shoot, `receiptId` should be `receiptID` right? Ugh... good catch!